### DISCUSSION TITLE
Future Changes

### DISCUSSION THREAD

**Discussion by elextr** [techee/geany-lsp#34] [Ideas]
To keep a discussion list of suggestions for future changes so they don't get lost.

@techee feel free to comment on the ideas of course.

1. Many of the lsp.conf global settings should be per server, eg hover may work fine for one language server but not another, and the formatting options as another example.
2. have a status bar icon that indicates the LSP is thinking, my test project is "larger", though by no means huge, so there are often delays in clangd responses.  Note thats the same with vscode, although its a bit quicker.

**Comment by techee** [comment]
> Many of the lsp.conf global settings should be per server, eg hover may work fine for one language server but not another, and the formatting options as another example.

RTFM :-) Basically all the settings from the `[all]` section can be overridden in the `[filetype]` section. The only exception is `enable_by_default` and `command_keybinding_num` which are valid only inside `[all]` (as described in the comment in the config file).

> have a status bar icon that indicates the LSP is thinking, my test project is "larger", though by no means huge, so there are often delays in clangd responses. Note thats the same with vscode, although its a bit quicker.

First, it's interesting it's slower than vscode - what LSP operations are you talking about? Is it autocompletion, goto, or what exactly?

Second, do plugins have some other means of accessing the status bar except `ui_set_statusbar` (which isn't the right one for this purpose)?

**Reply by elextr** [reply]
> First, it's interesting it's slower than vscode - what LSP operations are you talking about? Is it autocompletion, goto, or what exactly?

First open of a file to type highlighting being available, not that vscode is immediate either, but after that most things seem to be ok (limited testing so far, I only get a bit of time since my renovation is taking far too long and far too much attention, "what colour, matt or gloss, what taps, ... decisions decisions ;-S)

> do plugins have some other means of accessing the status bar

I thought there was a way of getting a progress bar in status, but maybe its not in the plugin API.

**Reply by techee** [reply]
>  (limited testing so far, I only get a bit of time since my renovation is taking far too long and far too much attention, "what colour, matt or gloss, what taps, ... decisions decisions ;-S)

Oh yeah, I know that. Condolences :-)

In any case, better not to spend too much time testing now and wait until we have the final API in Geany.

**Reply by techee** [reply]
> I thought there was a way of getting a progress bar in status, but maybe its not in the plugin API.

OK, yes, that's possible, I actually use it for progress messages from the server (like initial indexing).  I could show it also when I don't get a reply from the server for a long time, I just haven't seen such long response times (but I didn't test with any bigger C++ projects which I should try).
